File: backend/src/main/java/ba/sema/springbootvuedemo/services/CarService.java
package ba.sema.springbootvuedemo.services;

import ba.sema.springbootvuedemo.entities.Car;

import java.util.List;

public interface CarService
{
    List<Car> getCars();
    Car getCar(int id);
    void deleteCar(int id);
    Car createCar(Car car);
    Car updateCar(int id, Car car);
}

File: frontend/src/services/car-service.js
import Axios from "axios";


export default {
    getCars() {
        return Axios.get("car");
    },
    getCar(id) {
        return Axios.get(`car/${id}`);
    },
    deleteCar(id) {
        return Axios.delete(`car/${id}`);
    },
    createCar(carData) {
        return Axios.post("car", carData);
    },
    updateCar(id, carData) {
        return Axios.put(`car/${id}`, carData);
    }
};
